src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.container.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import React, { useEffect } from 'react';
+import type { ComponentType } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { EnrollmentPageComponent } from './EnrollmentPage.component';
+import type { EnrollmentPageStatus } from './EnrollmentPage.types';
+import { startFetchingEnrollmentPageInformation } from './EnrollmentPage.actions';
+
+export const EnrollmentPage: ComponentType<{||}> = () => {
+    const dispatch = useDispatch();
+
+    const enrollmentId: string =
+      useSelector(({ router: { location: { query } } }) => query.enrollmentId);
+    const enrollments: Object =
+      useSelector(({ enrollmentPage }) => enrollmentPage.enrollments);
+
+    useEffect(() => {
+        const enrollmentIsNotPreset = !(enrollments && enrollments.some(({ enrollment }) => enrollmentId === enrollment));
```

Comment:
probably dealt with later, but I see a loop forming if you query an enrollmentId that doesn't exist.